src/tagset.jl

```diff
@@ -3,28 +3,33 @@ const Tag = SmallString
 const maxTagLength = smallLength
 const MTagStorage = MSmallStringStorage # A mutable tag storage
 const IntTag = IntSmallString  # An integer that can be cast to a Tag
+const emptyIntTag = IntTag(0)
 const maxTags = 4
-const TagSetStorage = SVector{maxTags,IntTag}
-const MTagSetStorage = MVector{maxTags,IntTag}  # A mutable tag storage
+const vmaxTags = Val(maxTags)
```

Comment:
I think you have to literally write this as ` const vmaxTags = Val(4)` so then Julia sees the type parameter as a constant value `4`. Otherwise it has the same global variable/type instability problem as before.